### PR TITLE
release: split jobs, refactor

### PR DESCRIPTION
**Pull request by woodruffw**
This better separates our concerns (i.e. dist building vs. publishing).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,31 +5,52 @@ on:
...
-        attestations: true
+      - name: Upload distributions
+        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4
+        with:
+          name: distributions
+          path: dist/
+
+
+  publish:
+    name: publish release to PyPI
```
Comment (woodruffw) [inline .github/workflows/release.yml:37]:
```suggestion
    name: Publish Python 🐍 distributions 📦 to PyPI
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -5,31 +5,52 @@ on:
...
+          name: distributions
+          path: dist/
+
+
+  publish:
+    name: Publish Python 🐍 distributions 📦 to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/pypi-attestations
```
Comment (facutuesca) [inline .github/workflows/release.yml:41]: why do we need these?

**Review by facutuesca — APPROVED** [review]
LGTM! Just a small question

File: .github/workflows/release.yml (hunk 2 above)
Comment (woodruffw) [reply .github/workflows/release.yml:41]: We don't, it just makes the rendering in the actions status pane slightly prettier 🙂

**Pull request merged**